front/lib/api/assistant/conversation.ts

```diff
@@ -1655,3 +1669,39 @@ async function* streamRunAgentEvents(
     }
   }
 }
+
+async function isMessagesLimitReached({
+  owner,
+  plan,
+}: {
+  owner: WorkspaceType;
+  plan: PlanType;
+}): Promise<boolean> {
+  if (plan.limits.assistant.maxMessages === -1) {
+    return false;
+  }
+  /** Counting on workspace because user can be null */
+  const count = await Conversation.count({
```

Comment:
Probably need rework as per below

Reply:
Inner join returns all valid distinct couples (conversation, message)
I tested it before submitting the PR so it does what is expected and what we want

Reply:
Sorry just seeing this convo. I added a commit to rewrote the query. Good for you? 🙏🏻

Reply:
Closing discuss here if needed: https://github.com/dust-tt/dust/pull/2242#discussion_r1371918316
(I'll compare both query, if the first was better I will put it back with a comment to explain how it works and why it's normal we count from Conversation)